File: content-solutions/implementation/sim-rotate-matrix.java
import java.util.*;

class Solution {
    public int[][] solution(int[][] matrix, int k) {
        int[][] a = matrix;
        for (int t = 0; t < k % 4; t++) {
            int n = a.length, m = a[0].length;
            int[][] b = new int[m][n];
            for (int r = 0; r < n; r++)
                for (int c = 0; c < m; c++) b[c][n - 1 - r] = a[r][c];
            a = b;
        }
        return a;
    }
}

File: content-solutions/implementation/sim-rotate-matrix.py
def solution(matrix, k):
    a = matrix
    for _ in range(k % 4):
        n, m = len(a), len(a[0])
        b = [[0] * n for _ in range(m)]
        for r in range(n):
            for c in range(m):
                b[c][n - 1 - r] = a[r][c]
        a = b
    return a
